packages/framer-motion/src/animation/animators/js/index.ts

```diff
@@ -233,6 +227,12 @@ export function animateValue<V = number>({
             holdTime === null &&
             (playState === "finished" || (playState === "running" && done))
 
+        if (onUpdate) {
```

Comment:
By firing external listeners before we've determined whether the animation has finished, we can inadvertently end animations a frame early.